src/Routers/UsersRouter.js

```diff
@@ -289,6 +333,143 @@ export class UsersRouter extends ClassesRouter {
     }
     return Promise.resolve(success);
   }
+  encryptMFAKey(mfa, encryptionKey) {
+    try {
+      if (!encryptionKey) {
+        return mfa;
+      }
+      const algorithm = 'aes-256-gcm';
+      const encryption = crypto
+        .createHash('sha256')
+        .update(String(encryptionKey))
+        .digest('base64')
+        .substr(0, 32);
+      const iv = crypto.randomBytes(16);
+
+      const cipher = crypto.createCipheriv(algorithm, encryption, iv);
+
+      const encryptedResult = Buffer.concat([
+        cipher.update(mfa),
+        cipher.final(),
+        iv,
+        cipher.getAuthTag(),
+      ]);
+      return encryptedResult.toString('base64');
+    } catch (e) {
+      throw new Parse.Error(212, 'Invalid MFA token');
+    }
+  }
+  async decryptMFAKey(mfa, encryptionKey) {
+    try {
+      if (encryptionKey == null) {
+        return mfa;
+      }
+      const algorithm = 'aes-256-gcm';
+      const encryption = crypto
+        .createHash('sha256')
+        .update(String(encryptionKey))
+        .digest('base64')
+        .substr(0, 32);
+      const data = Buffer.from(mfa, 'base64');
+      const authTagLocation = data.length - 16;
+      const ivLocation = data.length - 32;
+      const authTag = data.slice(authTagLocation);
+      const iv = data.slice(ivLocation, authTagLocation);
+      const encrypted = data.slice(0, ivLocation);
+      const decipher = crypto.createDecipheriv(algorithm, encryption, iv);
+      decipher.setAuthTag(authTag);
+      return await new Promise((resolve, reject) => {
+        let decrypted = '';
+        decipher.on('readable', chunk => {
+          while (null !== (chunk = decipher.read())) {
+            decrypted += chunk.toString('utf8');
+          }
+        });
+        decipher.on('end', () => {
+          resolve(decrypted);
+        });
+        decipher.on('error', e => {
+          reject(e);
+        });
+        decipher.write(encrypted);
+        decipher.end();
+      });
+    } catch (err) {
+      throw new Parse.Error(212, 'Invalid MFA token');
+    }
+  }
+  async enableMFA(req) {
+    const mfaenabled = req.config.multiFactorAuth || {};
+    if (!mfaenabled.enabled) {
+      throw new Parse.Error(210, 'MFA is not enabled.');
+    }
+    if (!req.auth) {
+      throw new Parse.Error(101, 'Unauthorized');
```

Comment:
Requires Parse JS SDK reference for error code; check for possible inconsistency in error message for same error code.